lib/jekyll/page.rb

```diff
@@ -40,7 +40,11 @@ def initialize(site, base, dir, name)
       @base = base
       @dir  = dir
       @name = name
-      @path = site.in_source_dir(base, dir, name)
+      @path = if site.in_theme_dir(base) == base # we're in a theme
```

Comment:
@benbalter It essentially makes sure that we're normalising into the theme directory, rather than into the source directory, otherwise you might get `/home/user/work/home/user/.gems/[...]/theme.html` instead of `/home/user/.gems/[...]/theme.html`.

@parkr Have you tried this out with various symbolic links? I'm worried it might be a bit weird, depending on how `base` is set (I haven't checked, myself)